editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -571,20 +576,54 @@ export const UiJsxCanvas = React.memo<UiJsxCanvasPropsWithErrorCallback>((props)
       style={{
         all: 'initial',
       }}
+      id={CanvasContainerShadowRoot}
+      data-testid={CanvasContainerShadowRoot}
     >
-      <Helmet>{parse(linkTags)}</Helmet>
-      <RerenderUtopiaCtxAtom.Provider value={rerenderUtopiaContextValue}>
-        <UtopiaProjectCtxAtom.Provider value={utopiaProjectContextValue}>
-          <CanvasContainer
-            validRootPaths={rootValidPathsArray}
-            canvasRootElementElementPath={storyboardRootElementPath}
-          >
-            <SceneLevelUtopiaCtxAtom.Provider value={sceneLevelUtopiaContextValue}>
-              {StoryboardRoot}
-            </SceneLevelUtopiaCtxAtom.Provider>
-          </CanvasContainer>
-        </UtopiaProjectCtxAtom.Provider>
-      </RerenderUtopiaCtxAtom.Provider>
+      <ReactShadowRoot declarative={props.renderedInSSR}>
+        <EmotionWrapper>
+          <style>{`div,
+          span,
+          img,
+          ul,
+          li,
+          label {
+            box-sizing: border-box !important;
+          }`}</style>
+          <Helmet>{parse(linkTags)}</Helmet>
+          <RerenderUtopiaCtxAtom.Provider value={rerenderUtopiaContextValue}>
+            <UtopiaProjectCtxAtom.Provider value={utopiaProjectContextValue}>
+              <CanvasContainer
+                validRootPaths={rootValidPathsArray}
+                canvasRootElementElementPath={storyboardRootElementPath}
+              >
+                <SceneLevelUtopiaCtxAtom.Provider value={sceneLevelUtopiaContextValue}>
+                  {StoryboardRoot}
+                </SceneLevelUtopiaCtxAtom.Provider>
+              </CanvasContainer>
+            </UtopiaProjectCtxAtom.Provider>
+          </RerenderUtopiaCtxAtom.Provider>
+        </EmotionWrapper>
+      </ReactShadowRoot>
+    </div>
+  )
+})
+
+const EmotionWrapper = React.memo(({ children }: { children?: React.ReactNode | undefined }) => {
```

Comment:
Might be worth putting the explanation from the PR description against this for future people.

Reply:
Good shout!